#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("natural_frequency", double_t, 0, "TODO", 1., 0.1, 10.)
gen.add("damping_ratio", double_t, 0, "TODO", 1., 0.7, 1.)
gen.add("prediction_horizon", double_t, 0, "TODO", 1., 0.1, 3.)
gen.add("prediction_steps", int_t, 0, "TODO", 10, 1, 30)
gen.add("rotation_decay", double_t, 0, "TODO", 0.2, 0., 1.)
gen.add("angular_velocity_decay", double_t, 0, "TODO", 0., 0., 1.)
gen.add("rotation_weight", int_t, 0, "TODO", 100, 1, 100)
gen.add("angular_velocity_weight", int_t, 0, "TODO", 1, 1, 100)
gen.add("thrust_force_weight", int_t, 0, "TODO", -3, -6, 0)
gen.add("thrust_force_rate_weight", int_t, 0, "TODO", -3, -6, 0)

exit(gen.generate("multirotor_controller", "controller", "Controller"))
